src/main/java/uk/gov/hmcts/reform/translate/helper/DictionaryMapper.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.translate.helper;
+
+import lombok.val;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.translate.data.DictionaryEntity;
+import uk.gov.hmcts.reform.translate.data.TranslationUploadEntity;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+@Component
+public class DictionaryMapper {
+
+    public DictionaryEntity modelToEntityWithTranslationUploadEntity(final Map.Entry<String, String> currentPhrase,
+                                                                     final String currentUserId) {
+
+        val dictionaryEntity = modelToEntity(currentPhrase);
+        val translationUploadEntity = new TranslationUploadEntity();
+        translationUploadEntity.setUserId(currentUserId);
```

Comment:
instead of having to pass in `currentUserId` here could we not just call

`SecurityUtils.getUserId()` ?

Reply:
well , that could lead in to overhead as return getUserInfo().getUid() in SecurityUtils end up calling  UserInfo userInfo = idamRepository.getUserInfo(getUserToken());